File: Web/Source/Toffees.Web/ClientApp/src/app/auth/client-authorization-credentials.ts
export class ClientAuthorizationCredentials {
  clientId: string;
  clientSecret: string;
  grantType: string;
  scope: string[];

  constructor(clientId: string, clientSecret: string, grantType: string, scope: string[]) {
    this.clientId = clientId;
    this.clientSecret = clientSecret;
    this.grantType = grantType;
    this.scope = scope;
  }
}

File: Web/Api/Models/Auth/ClientAuthorizationCredentials.cs
using Newtonsoft.Json;

namespace Toffees.Web.Api.Models.Auth
{
    public class ClientAuthorizationCredentials
    {
        [JsonProperty("clientId")]
        public string ClientId { get; set; }

        [JsonProperty("clientSecret")]
        public string ClientSecret { get; set; }

        [JsonProperty("grantType")]
        public string GrantType { get; set; }

        [JsonProperty("scope")]
        public string[] Scope { get; set; }
    }
}
